spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/CrudMethodMetadata.java

```diff
@@ -85,4 +86,11 @@ public interface CrudMethodMetadata {
 	 * @since 1.9
 	 */
 	Method getMethod();
+
```

Comment:
`JpaRepositoryImplementation` provides already customization options for repository implementations. What if we evolve that concept by moving `setRepositoryMethodMetadata` and `setEscapeCharacter` methods into a new interface, that would also accept `ProjectionFactory` and let both, `JpaRepositoryImplementation` and `QuerydslJpaPredicateExecutor` implement that interface?

Reply:
worth exploring - also in the light of having to deal with a similar arrangement in other store modules.